### PR TITLE
[+] Related Data - Delete records directly from a hasMany listing

### PR DESCRIPTION
**Pull request by root-io**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by root-io** [comment]
[(5) Users want to delete records directly from a hasMany listing](https://trello.com/c/iQNkAxsm/2322-5-users-want-to-delete-records-directly-from-a-hasmany-listing)

File: services/has-many-dissociator.js [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
+          return record['remove' + _.capitalize(params.associationName)](
+            associatedIds);
+        } else {
+          return null;
+        }
+      })
+      .then(function () {
+        if (isDelete) {
+          return association.destroy({
+            where: {id: { [OPERATORS.IN]: associatedIds }}
```
Comment (ArnaudValensi) [inline services/has-many-dissociator.js]: Some spaces missing: 
```
where: { id: { [OPERATORS.IN]: associatedIds } }
```

We will be add babel and eslint someday

File: services/has-many-dissociator.js [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+          });
+        } else {
+          removeAssociation = true;
+        }
+        if (removeAssociation) {
+          return record['remove' + _.capitalize(params.associationName)](
+            associatedIds);
+        } else {
+          return null;
+        }
```
Comment (ArnaudValensi) [inline services/has-many-dissociator.js]: You can ommi the else and just return:
```
        if (removeAssociation) {
          return record['remove' + _.capitalize(params.associationName)](
            associatedIds);
        }
        return null;
```

File: test/index.js [hunk 3]
```diff
@@ -1073,5 +1109,117 @@ var HasManyGetter = require('../services/has-many-getter');
...
+                  .then(function (userTeam) {
+                    expect(userTeam).to.be.null;
+                    done();
+                  });
+              })
+              .catch(done);
+          });
+        });
+      });
+    });
```
Comment (ArnaudValensi) [inline test/index.js]: You can add tests which try to delete and dissociate on a record which is not an relationship.

File: services/has-many-dissociator.js [hunk 4]
```diff
@@ -1,22 +1,46 @@
 'use strict';
 var _ = require('lodash');
+var Operators = require('../utils/operators');
 var ErrorHTTP422 = require('./errors').ErrorHTTP422;
 
-function HasManyAssociator(model, association, opts, params, data) {
+function HasManyDissociator(model, association, options, params, data) {
```
Comment (ArnaudValensi) [inline services/has-many-dissociator.js]: You should check if `association` exists and return a 404 if does not. Here or in forest-expres

**Pull request merged**